FantasmoSDK/Classes/Filters/FMCompoundFrameQualityFilter.swift

```diff
@@ -0,0 +1,97 @@
+//
+//  FMInputQualityFilter.swift
+//  FantasmoSDK
+//
+//  Created by lucas kuzma on 3/30/21.
+//
+
+import ARKit
+
+// MARK:-
+
+/// Stateful filter for choosing the frames which are acceptable to localize against.
+/// Frames are expected to be passed sequentially to the `accepts(_:)` method in the proper order.
+/// If it is necessary to process a new sequence of frames, then `startOrRestartFiltering()` must be invoked.
+class FMCompoundFrameQualityFilter {
+    
+    /// We use `clock()` rathern than `Date()` as it is likely faster. Some hint at this can be found at https://bit.ly/3vExXcZ
+    private var timestampOLastAcceptedFrame: clock_t?
+    
+    /// The number of seconds after which we forcibly accept a frame.
+    private var acceptanceThreshold = 6.0
+    
+    /// Filter collection, in order of increasing computational cost
+    private let filters: [FMFrameFilter] = [
+        FMCameraPitchFilter(),
+        FMMovementFilter(),
+        FMBlurFilter(),
+    ]
+
+    /// Invoke this method when it is needed to start validating a new sequence of frames.
+    /// Impl details: Invoking this method will ensure that first frame on the new sequence will not be force approved without assessing for quality.
+    func startOrRestartFiltering() {
+        timestampOLastAcceptedFrame = nil
+    }
+    
+    /// Indicate whether passed `frame` should be used for the localization.
+    /// Frame is assessed for quality by various aspects and is accepted without any assessment if the last successfully accepted
+    /// frame was passed long ago.
+    /// If it is needed to start working with new sequence of frames then invoke `startOrRestartFiltering()` or create new instance
+    /// of this class.
+    func accepts(_ frame: ARFrame) -> FMFrameFilterResult {
+        if shouldForciblyAccept(frame) {
+            timestampOLastAcceptedFrame = clock()
+            print("***************************1")
```

Comment:
Is this some debug info that should be removed?

Reply:
Yes, thanks. I have fixed this.